### PR TITLE
SmokeTests and automation base

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- New feature

### Description

Currently, there were no option to run some kind of system tests against registries. This PR add option to run specific version of builded registries (from jar file) together with kafka and simulate real deployment of both systems.

This PR also add some basic tests for artifacts, metadata, versions and rules.

There are still some things which should be done:
- [ ] An option to run tests against registries on specific url (it's partly completed)
- [ ] Fix logging in the maven (logs are currently available only in run triggered from InteliJ)
- [ ] Run these tests as part of CircleCI builds

### Checklist
- [ ] Write tests
- [ ] Make sure all tests pass

### CODE REVIEW

File: tests/src/main/java/io/apicurio/tests/utils/HttpUtils.java [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+
+import static io.restassured.RestAssured.given;
+
+public class HttpUtils {
```
Comment (alesj) [inline tests/src/main/java/io/apicurio/tests/utils/HttpUtils.java]: Why don't you just use the Registry client we already have impled?

**Comment by EricWittmann** [comment]
I apologize, but can you close this PR and re-open it against the `1.0.x` branch?  I forgot to merge this work before I made that branch. :(

**Comment by EricWittmann** [comment]
@Frawless ^^^^

**Comment by Frawless** [comment]
@EricWittmann sure, I will do that when I find some time for registries again

File: tests/src/main/java/io/apicurio/tests/utils/HttpUtils.java (hunk 1 above)
Comment (Frawless) [reply tests/src/main/java/io/apicurio/tests/utils/HttpUtils.java]: I didn't know we have it, I will change it :)

File: tests/src/test/java/io/apicurio/tests/BaseIT.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+public abstract class BaseIT implements TestSeparator, Constants {
+    static final Logger LOGGER = LoggerFactory.getLogger(BaseIT.class);
+
+    private static KafkaFacade kafkaCluster = new KafkaFacade();
+    private static RegistryFacade registries = new RegistryFacade();
+
+    @BeforeAll
+    static void beforeAll() throws TimeoutException {
+        kafkaCluster.start();
```
[the remark this answers is not in the document]
Comment (Frawless) [reply tests/src/test/java/io/apicurio/tests/BaseIT.java]: Guess we can put it into some share package and re-use it :)

**Pull request closed without merging**